enhancements/microshift/advertise-version-details.md

```diff
@@ -0,0 +1,303 @@
+---
+title: advertise-version-details
+authors:
+  - "@dhellmann"
+reviewers: # Include a comment about what domain expertise a reviewer is expected to bring and what area of the enhancement you expect them to focus on. For example: - "@networkguru, for networking aspects, please look at IP bootstrapping aspect"
+  - "@spadgett, console team"
+  - "@wking, upgrades team"
+  - "@LalatenduMohanty, upgrades team"
+  - "@soltysh, workloads team"
+approvers:
+  - "@sdodson"
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - None
+creation-date: 2022-07-29
+last-updated: 2022-07-29
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/USHIFT-233
+  - https://bugzilla.redhat.com/show_bug.cgi?id=1850656
+---
+
+# Advertise Version Details
+
+## Summary
+
+We have several use cases for allowing authenticated users to discover
+whether they are communicating with OCP or MicroShift, including the
+version of the product. We already have the ClusterVersion resource
+for OCP, but that requires special privileges to read and is not
+present in a MicroShift deployment. This enhancement describes a
+standard that both platforms can use to share platform and version
+information with API callers.
+
+## Motivation
+
+Users configuring applications to deploy on different versions of
+OpenShift may need to adjust their configuration to work differently
+on OCP and MicroShift. Ideally this is not needed, but in practice it
+is useful because of different APIs being available and different
+constraints being in place. It is possible for a cluster admin to read
+the ClusterVersion resource in an OCP cluster to discover that it is
+OCP and what version is running. Limiting ourselves to this approach has two
+drawbacks. First, not all console users have cluster admin privileges,
+which means that the console cannot show them the cluster version
+details. Second, the ClusterVersion API is not present in a MicroShift
+deployment, so instead [we created a
+ConfigMap](https://github.com/openshift/microshift/pull/776) in the
+`openshift-public` namespace containing some basic version
```

Comment:
openshift/microshift#776 used `microshift-version` in `kube-public`.  Later in this enhancement, you mention `version` in `openshift`.  This line is the only ref I see to `openshift-public`, and we don't seem to have a namespace like that in CI.  Probably pick either `openshift` or `kube-public` here, depending on whether you want to focus on your proposal or the existing MicroShift implementation, when motivating the proposal.  Or just say "in a well known namespace" or something similarly generic.

Reply:
Thanks, that should be `kube-public`.